### DISCUSSION TITLE
How to iterate over a pcap file without extracting all the frames?

### DISCUSSION THREAD

**Discussion by cabezadevaca** [JarryShaw/PyPCAPKit#127] [General]
Hi, I have been trying to extract frames from ~500M PCAP file. Python consumes around 10G of memory causing process swapping and killing after all. Is there a way to have a subset of frames extracted and iterated?

Thanks!
Rustem

**Comment by JarryShaw** [comment]
There is a possible solution to set `store=False` when calling `extract` (or equivalent) and then set `verbose=func` where `func` can be used as a filter function to determine such frame should be extracted (stored) or not.

Another approach is to set `auto=False` when calling `extract` (or equivalent), which will return an iterator instance allowing you to extract each frame when called, and manually iterate through each frame upon iterator's entry.
